src/adapter/src/catalog.rs

```diff
@@ -3532,8 +3532,17 @@ impl Catalog {
             );
         }
         let cluster = self.resolve_cluster(session.vars().cluster())?;
-        if cluster.linked_object_id.is_some() {
-            coord_bail!("cannot execute queries on linked clusters");
+        // Disallow queries on storage clusters. There's no technical reason for
+        // this restriction, just a philosophical one: we want all crashes in
+        // a storage cluster to be the result of sources and sinks, not user
+        // queries.
+        if cluster.bound_objects.iter().any(|id| {
```

Comment:
While you're on this, I was wondering why these use `all()` instead of `any()`: https://github.com/MaterializeInc/materialize/blob/c88a9e8d926ae39df36282d79623e2131372cca0/src/sql/src/plan/statement/ddl.rs#L3361

Reply:
Oh yeah, I saw that comment! I think it is equivalent, right? Because we never allow you to mix-n-match. If you think `any` is clearer happy to switch, but hopefully this code doesn't last long because we'll allow you to mix-n-match soon. 🤞🏽

Reply:
They are equivalent, yes, because we don't allow mixing. `any` seemed clearer to me, but I don't have a strong opinion!